packages/@o3r/rules-engine/builders/rules-engine-extractor/helpers/rules-engine.extractor.ts

```diff
@@ -293,21 +292,23 @@ export class RulesEngineExtractor {
           }
 
           declaration.type.typeArguments?.forEach((argType, idx) => {
-            const operand = idx === 0 ? 'leftOperand' : 'rightOperand';
-            const operandObject = operator[operand]!;
-
-            operandObject.nbValues = this.getTypeNbValue(argType);
-
-            const typeNode = this.extractTypeNode(argType);
-            if (typeNode.kind === ts.SyntaxKind.UndefinedKeyword) {
-              delete operator[operand];
-            } else {
-              const typeData = this.extractComplexTypeData(typeNode, source, operandObject.nbValues);
-              operandObject.nbValues = typeData.nbValue;
-              operandObject.types = typeData.types;
-              if (operandObject.types.some((type) => type !== 'unknown' && allSupportedTypes.indexOf(type) === -1)) {
-                this.logger.warn(`Operator ${operator.id} has an unsupported type in operand ${operand}: "${operandObject.types.join(', ')}"`);
-                operandObject.types = ['unknown'];
+            if (idx <= 1) {
+              const operand = idx === 0 ? 'leftOperand' : 'rightOperand';
```

Comment:
(not mandatory but cooler :D :)

```suggestion
              const operands = ['leftOperand', 'rightOperand'];
              const operand = operands[idx];
```
(even better if the array in externalize)